plugins/sources/telegraf/provider.go

```diff
@@ -0,0 +1,206 @@
+package telegraf
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf/plugins/inputs/procstat"
+
+	"github.com/golang/glog"
+
+	"github.com/influxdata/telegraf"
+	telegrafInputs "github.com/influxdata/telegraf/plugins/inputs"
+	wf "github.com/wavefronthq/wavefront-kubernetes-collector/internal/metrics"
+)
+
+// NewProvider creates a Telegraf source
+func NewProvider(uri *url.URL) (wf.MetricsSourceProvider, error) {
+
+	for _, pair := range os.Environ() {
+		glog.Infof("env: %v", pair)
+	}
+
+	glog.Infof("[telegraf.NewProvider] - inputs: %v -------------", telegrafInputs.Inputs)
+
+	var sources []wf.MetricsSource
+	for name, creator := range telegrafInputs.Inputs {
```

Comment:
Can we make this configurable? If `telegraf:''` we can default to including all inputs. But also provide user an option to specify what they want. We'd need some kind of mapping/sanitizing for the input name.